inference-engine/src/transformations/src/transformations/common_optimizations/shuffle_channels_fusion.cpp

```diff
@@ -0,0 +1,138 @@
+            expected_transpose_values.push_back(4);
+        }
+
+        is_transpose_valid &= (expected_transpose_values == transpose_constant_values);
+    }
+
+    // y = reshape(x'', [N, C, H, W])
+    bool is_reshape_after_valid = (shape_input == shape_reshape_after);
+
+    is_transformation_valid &= is_reshape_before_valid & is_transpose_valid & is_reshape_after_valid;
+    return is_transformation_valid;
+}
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ShuffleChannelsFusion, "ShuffleChannelsFusion", 0);
+
+ngraph::pass::ShuffleChannelsFusion::ShuffleChannelsFusion(const bool reshape_constants_check) {
+    MATCHER_SCOPE(ShuffleChannelsFusion);
+    auto input0 = std::make_shared<pattern::op::Label>(element::f32);
+    auto input1 = std::make_shared<pattern::op::Label>(element::i64);
+    auto input2 = std::make_shared<pattern::op::Label>(element::i64);
+    auto input3 = std::make_shared<pattern::op::Label>(element::i64);
+    auto reshape_before = std::make_shared<ngraph::opset3::Reshape>(input0, input1, false);
+    auto transpose = std::make_shared<ngraph::opset3::Transpose>(reshape_before, input2);
+    auto reshape_after = std::make_shared<ngraph::opset3::Reshape>(transpose, input3, false);
+
+    ngraph::matcher_pass_callback callback = [this, reshape_constants_check](pattern::Matcher& m) {
+        auto reshape_after = std::dynamic_pointer_cast<ngraph::opset3::Reshape>(m.get_match_root());
+        if (!reshape_after) {
+            return false;
+        }
+
+        auto transpose = std::dynamic_pointer_cast<ngraph::opset3::Transpose>(reshape_after->input_value(0).get_node_shared_ptr());
+        if (!transpose || transpose->get_output_target_inputs(0).size() != 1) {
+            return false;
+        }
+
+        auto reshape_before = std::dynamic_pointer_cast<ngraph::opset3::Reshape>(transpose->input_value(0).get_node_shared_ptr());
+        if (!reshape_before || reshape_before->get_output_target_inputs(0).size() != 1) {
+            return false;
+        }
+
+        auto reshape_after_constant = std::dynamic_pointer_cast<ngraph::opset3::Constant>(reshape_after->input_value(1).get_node_shared_ptr());
+        auto transpose_constant = std::dynamic_pointer_cast<ngraph::opset3::Constant>(transpose->input_value(1).get_node_shared_ptr());
+        auto reshape_before_constant = std::dynamic_pointer_cast<ngraph::opset3::Constant>(reshape_before->input_value(1).get_node_shared_ptr());
+        if (!transpose_constant || !reshape_before_constant || !reshape_after_constant) {
+            return false;
+        }
+
+        if (reshape_constants_check) {
+            auto reshape_before_values = reshape_before_constant->cast_vector<int64_t>();
+            auto reshape_after_values = reshape_after_constant->cast_vector<int64_t>();
+            for (size_t i = 0; i < reshape_before_values.size(); ++i) {
+                if ((reshape_before_values[i] == -1) || (reshape_before_values[i] == 0 && i != 0)) {
```

Comment:
Why we prohibit `0` value? Please explain.

Reply:
This is an unnecessary check. I removed it. Thanks!